### ISSUE TITLE
Remove aria-required-children for role="tree"

### ISSUE THREAD

**Issue by robpaveza** [dequelabs/axe-core#1444]
**Expectation**:

```html
<ol role="tree"></ol>
```

should generate no errors.

**Actual**: It generates the following error:

> [aria-required-children](https://dequeuniversity.com/rules/axe/3.1/aria-required-children?application=axeAPI): Certain ARIA roles must contain particular children  [WCAG 1.3.1](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships)

**Motivation**: A tree that has no leaves is still a tree, and therefore `role="treeitem"` should not be a required child of an element of `role="tree"`.

**axe-core version**: 3.0.0-alpha1, 3.1.1

**Comment by smhigley** [comment]
We ran into this issue as well in a product that allows user interaction to insert tree items. The empty `role="tree"` exists before any tree items have been inserted.

There are examples of empty trees existing on desktop applications in Windows applications, so it's not a completely unknown pattern. An example might be in something like this: https://www.lightningdesignsystem.com/components/dueling-picklist/ where instead of listboxes there are selectible tree views, and all items are removed from one side.

**Comment by WilcoFiers** [comment]
I agree. I think generally the aria-required-children rule isn't quite right and should probably be more of an "allowed children" type rule, similar to what we've done with the list rules.

Thank you for reporting. We'll definitely pick this up.

**Comment by straker** [comment]
@WilcoFiers how exactly would you like to handle this? It seems related to #874 and #383

**Comment by WilcoFiers** [comment]
@straker The aria-required-children check has an option for `reviewEmpty`, which when we put a role in there, puts the results into "needs review" rather than failing it. I think it should have the following items in it, so that they are put into needs review:

- doc-bibliography
- doc-endnotes
- list
- listbox
- table
- grid
- treegrid
- tree
- group
- rowgroup
- tablist


That would make only the following roles fail if they are left empty:
- menu
- menubar
- row

**Comment by chandana7393** [comment]
Verified for the required test file, not seeing any voilations, but observed "Needs review" issue.
(Attached the screenshot) 
![need_review_no_children](https://user-images.githubusercontent.com/45285854/58472527-0f01ce00-8164-11e9-8126-0570efc334d2.PNG)
**Tested Environment:**
Axe-coconut - 3.8.0.14704v
Chrome - 74v
OS - Windows 10 64 bit
